metadata_generator.py

```diff
@@ -0,0 +1,159 @@
+"""Metadata generator module."""
+
+import argparse
+import hashlib
+import json
+import os
+from datetime import datetime
+
+from mcap.reader import make_reader
+
+DEFAULT_PATH = '/recorded_datasets/edinburgh'
+
+
+def read_mcap_file(mcap_file_path):
+    """
+    Read an MCAP file and extract metadata.
+
+    Parameters:
+    mcap_file_path (str): The path to the MCAP file.
+
+    Returns:
+    dict: A dictionary containing the duration and topic message counts.
+    """
+    topic_message_counts = {}
+    start_time = None
+    end_time = None
+
+    with open(mcap_file_path, 'rb') as f:
+        reader = make_reader(f)
+        for schema, channel, message in reader.iter_messages():
+            if start_time is None:
+                start_time = message.log_time
+            end_time = message.log_time
+
+            topic = channel.topic
+            if topic not in topic_message_counts:
+                topic_message_counts[topic] = 0
+            topic_message_counts[topic] += 1
+
+    duration = end_time - start_time
+    duration_seconds = duration / 1e9  # Convert nanoseconds to seconds
+
+    result = {
+        'duration': f'{duration_seconds: .0f}s',
+        'topics': topic_message_counts,

```

Comment:
Can we also define the msg type name or schema ?

For instance
`msg_type: sensor_msgs/msg/Imu`

Reply:
Added